admin-dev/themes/new-theme/js/components/multistore-header.ts

```diff
@@ -79,6 +80,43 @@ const initMultistoreHeader = () => {
   }
 };
 
+const initContextualNotification = () => {
+  const configKeyShopPrefix = 'missing-color-shop-';
+  const configKeyGroupPrefix = 'missing-color-group-';
+  const multistoreHeader = document.querySelector('.header-multishop');
+
+  if (multistoreHeader === null || !(multistoreHeader instanceof HTMLElement) || !multistoreHeader.dataset.headerColorNotification) {
+    return;
+  }
+
+  const notificationMessage = multistoreHeader.dataset.headerColorNotification;
+
+  if (notificationMessage.length <= 0) {
+    return;
+  }
+
+  // make localstorage key for this context
+  const contextualNotification = new ContextualNotification();
+  let notificationKey = configKeyGroupPrefix + multistoreHeader.dataset.groupId;
+
+  if (multistoreHeader.hasAttribute('data-shop-id')) {
+    notificationKey = configKeyShopPrefix + multistoreHeader.dataset.shopId;
+  }
+
+  // check if key exists, if yes: display or not depending on given value
+  const configValue = contextualNotification.getItem(notificationKey);
+
+  if (configValue === true || configValue === null) {
+    contextualNotification.displayNotification(multistoreHeader.dataset.headerColorNotification, notificationKey);
+  }
+
+  // if the config doesn't exist, we set it to true
+  if (configValue === null) {
+    contextualNotification.setItem(notificationKey, true);
+  }
```

Comment:
looks like I'm reading the same code as multistore-config-field

I see the few differences and I don't see how to mutualize it without making it dirty

So let it as it is

Reply:
how is this the same as `multistore-config-field.ts`? 🤔